core/drivers/qti/ramblur/ramblur_pimem.c

```diff
@@ -0,0 +1,260 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2025, Qualcomm Technologies Inc. and/or its subsidiaries.
+ */
+
+#include <assert.h>
+#include <initcall.h>
+#include <io.h>
+#include <kernel/panic.h>
+#include <mm/core_memprot.h>
+#include <ramblur_pimem.h>
+#include <ramblur_pimem_hwio.h>
+#include <string.h>
+
+#define ANTIROLLBACK	(UINT32_C(1) << \
+	RAMBLUR_PIMEM_WIN0_ALGORITHM_CONFIG_WIN_ANTIROLLBACK_ENABLE_SHFT)
+#define INTEGRITY	(UINT32_C(1) << \
+	RAMBLUR_PIMEM_WIN0_ALGORITHM_CONFIG_WIN_INTEGRITY_ENABLE_SHFT)
+#define CONFIDENTIALITY	(UINT32_C(1) << \
+	RAMBLUR_PIMEM_WIN0_ALGORITHM_CONFIG_WIN_CONFIDENTIALITY_ENABLE_SHFT)
+
+static vaddr_t ramblur_va;
+
+static inline uint16_t in_word(vaddr_t addr)
+{
+	return io_read16(addr + ramblur_va);
+}
+
+static inline uint32_t in_dword(vaddr_t addr)
+{
+	return io_read32(addr + ramblur_va);
+}
+
+static inline uint32_t in_dword_masked(vaddr_t addr, uint32_t mask)
+{
+	uint32_t val = in_dword(addr);
+
+	return val & mask;
+}
+
+static inline void out_dword(vaddr_t port, uint32_t val)
+{
+	io_write32(port + ramblur_va, val);
+	dsb();
+}
+
+static inline void out_dword_masked_ns(vaddr_t io, uint32_t mask, uint32_t val,
+				       uint32_t current_reg_content)
```

Comment:
Why do you need to these wrappers here and above? `io_mask32()` can be used here instead directly.

Reply:
sorry what duplication? the header uses macros that the program uses as well. this sort of encapsulation is common in Op-tee (check i2c drivers for example)